Simulation Interfaces Docs update (#2720)
o3de/o3de.org: content/docs/user-guide/interactivity/robotics/simulation-interfaces.md

| `SimulationInterfaces/KeyboardTransitions/StoppedToPlaying` | Set the keyboard key that will change simulation state from Stopped to Playing e.g., `keyboard_key_alphanumeric_R` |
| `SimulationInterfaces/KeyboardTransitions/PausedToPlaying`  | Set the keyboard key that will change simulation state from Paused to Playing e.g., `keyboard_key_alphanumeric_P`  |
| `SimulationInterfaces/KeyboardTransitions/PlayingToPaused`  | Set the keyboard key that will change simulation state from Playing to Paused e.g., `keyboard_key_alphanumeric_P`  |

> | `ROS2SimulationInterfaces/<ServiceType>`                    | Change default service name, if empty string is pass service is disabled                                           |

Reviewer: ```suggestion
| `ROS2SimulationInterfaces/<ServiceType>`                    | Change default service name (service is disabled if empty string is passed)                                        |
```

- SPAWNING
- DELETING
- NAMED_POSES
- POSE_BOUNDS
- ENTITY_TAGS
- ENTITY_BOUNDS
- ENTITY_BOUNDS_BOX
- ENTITY_CATEGORIES 
- ENTITY_STATE_GETTING
- ENTITY_STATE_SETTING
- ENTITY_INFO_GETTING
- ENTITY_INFO_SETTING
- SPAWNABLES
- SIMULATION_RESET
- SIMULATION_RESET_TIME
- SIMULATION_RESET_STATE
- SIMULATION_RESET_SPAWNED
- SIMULATION_STATE_GETTING
- SIMULATION_STATE_SETTING
- SIMULATION_STATE_PAUSE
- STEP_SIMULATION_SINGLE
- STEP_SIMULATION_MULTIPLE
- STEP_SIMULATION_ACTION
- WORLD_LOADING
- WORLD_UNLOADING
- WORLD_INFO_GETTING
- AVAILABLE_WORLDS

### SpawnEntity service

> The `SpawnEntity` _service_ lets you add simulated _spawnables_ into the simulation environment. List of available _spawnables_ gan be obtained with the `GetSpawnables` _service_.

Reviewer: ```suggestion
The `SpawnEntity` _service_ lets you add simulated _spawnables_ into the simulation environment. List of available _spawnables_ can be obtained with the `GetSpawnables` _service_.
```

ROS 2 _service_ definition: [SpawnEntity.srv](https://github.com/ros-simulation/simulation_interfaces/tree/main/srv/SpawnEntity.srv) \
Default _service_ name: `/spawn_entity` \
O3DE EBus: `SimulationInterfaces::SimulationEntityManagerRequests::SpawnEntity`

If you want to spawn a new object into the simulation environment, you need to know the valid URI of the _spawnable_, e.g.,`product_asset:///prefabs/robot/foorobot.spawnable`. `Simulation Entities Manager` will find corresponding Asset ID based on the URI.

**Note:** Spawning assets from the file system e.g., `spawnable://home/username/robots/FooRobot.spawnables` is not supported.

### DeleteEntity service

The `DeleteEntity` _service_ lets you despawn the previously spawned entities.

ROS 2 _service_ definition: [DeleteEntity.srv](https://github.com/ros-simulation/simulation_interfaces/tree/main/srv/DeleteEntity.srv) \
Default _service_ name: `/delete_entity` \
O3DE EBus: `SimulationInterfaces::SimulationEntityManagerRequests::DeleteEntity`

> **Note:** This mechanism allows you only to delete entities which were spawned by the `SimulationInterfaces` Gem.

Reviewer: ```suggestion
**Note:** This mechanism allows you to delete only entities which were previously spawned with the `SimulationInterfaces` interface.
```

O3DE EBus: `SimulationInterfaces::SimulationManagerRequests::StepSimulation`

**Note:** The simulator has to be paused before calling this _action_. Otherwise, `SimulateSteps` goal will be accepted and fail immediately.

### LoadWorld service

> The `LoadWorld` _service_ allows to load level in simulation. Calling `LoadWorld` _service_ with already loaded level, unloads old one and loads requested level. 

Reviewer: ```suggestion
The `LoadWorld` _service_ loads levels into the simulation. Calling the _service_ with a loaded level unloads it before loading the requested level. 
```

ROS 2 _service_ definition: [LoadWorld.srv](https://github.com/ros-simulation/simulation_interfaces/blob/main/srv/LoadWorld.srv) \
Default _service_ name: `/load_world` \
O3DE EBus: `SimulationInterfaces::LevelManager::LoadWorld`

**Note** The simulator currently doesn't support loading levels from resource string.

### UnloadWorld service

> The `UnloadWorld` _service_ allows to unload current level in simulation. 

Reviewer: ```suggestion
The `UnloadWorld` _service_ unloads the current level from the simulation. 
```

ROS 2 _service_ definition: [UnloadWorld.srv](https://github.com/ros-simulation/simulation_interfaces/blob/main/srv/UnloadWorld.srv) \
Default _service_ name: `/unload_world` \
O3DE EBus: `SimulationInterfaces::LevelManager::UnloadWorld`

### GetCurrentWorld service

> The `GetCurrentWorld` _service_ allows to get data about currently loaded level.

Reviewer: ```suggestion
The `GetCurrentWorld` _service_ publishes information about the level that is currently loaded into the simulation.
```

ROS 2 _service_ definition: [GetCurrentWorld.srv](https://github.com/ros-simulation/simulation_interfaces/blob/main/srv/GetCurrentWorld.srv) \
Default _service_ name: `/get_current_world` \
O3DE EBus: `SimulationInterfaces::LevelManager::GetCurrentWorld`

### GetAvailableWorlds service

> The `GetAvailableWorlds` _service_ allows to list all available levels

Reviewer: ```suggestion
The `GetAvailableWorlds` _service_ lists all available levels.
```

ROS 2 _service_ definition: [GetAvailableWorlds.srv](https://github.com/ros-simulation/simulation_interfaces/blob/main/srv/GetAvailableWorlds.srv) \
Default _service_ name: `/get_available_worlds` \
O3DE EBus: `SimulationInterfaces::LevelManager::GetAvailableWorlds`

> **Note** The simulator currently doesn't support searching in online resources

Reviewer: ```suggestion
**Note** The simulator doesn't support searching for online resources.
```
